docker/registry/acr.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package registry
+
+import (
+	"encoding/base64"
+	"net/http"
+	"strings"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/docker"
+)
+
+type acr struct {
+	*baseClient
+}
+
+func newACR(repoDetails docker.ImageRepoDetails, transport http.RoundTripper) RegistryInternal {
```

Comment:
rename to something easier to identitfy like newAzureACR or newAzureContainerRegistry